nix: update packaging to Qt 6

imincik opened the pull request:
## Description

Update QGIS Nix packaging to Qt 6.

PR is ready for review and testing, QGIS looks working nicely:

```
nix build github:qgis/QGIS/pull/62574/merge#qgis
./result/bin/qgis
```

or just

```
nix run github:qgis/QGIS/pull/62574/merge#qgis
```

<img width="1925" height="1080" alt="screen-imincik-2025-08-25-16:00:28" src="https://github.com/user-attachments/assets/8ba822b2-f8ed-4a37-8a6d-83eda66b917b" />

/cc @timlinux @autra @m-kuhn 


<!--
  BEFORE HITTING SUBMIT -- Please BUILD AND TEST your changes thoroughly. This is YOUR responsibility! Do NOT rely on the QGIS code maintainers to do this for you!!

  IMPORTANT NOTES FOR FIRST TIME CONTRIBUTORS
  ===========================================

  Congratulations, you are about to make a pull request to QGIS! To make this as easy and pleasurable for everyone, please take the time to read these lines before opening the pull request.

  Include a few sentences describing the overall goals for this pull request (PR). If applicable also add screenshots or - even better - screencasts.
  Include both: *what* you changed and *why* you changed it.

  If this is a pull request that adds new functionality which needs documentation, give an especially detailed explanation.
  In this case, start with a short abstract and then write some text that can be copied 1:1 to the documentation in the best case.

  Also mention if you think this PR needs to be backported. And list relevant or fixed issues.

------------------------

  Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by checking the following list.
  Feel free to ask in a comment if you have troubles with any of them.

  - Commit messages are descriptive and explain the rationale for changes.

  - Commits which fix bugs include `Fixes #11111` at the bottom of the commit message. If this is your first pull request and you forgot to do this, write the same statement into this text field with the pull request description.

  - New unit tests have been added for relevant changes

  - you have read the QGIS Developer Guide (https://docs.qgis.org/testing/en/docs/developers_guide/index.html) and your PR complies with its QGIS Coding Standards
-->

imincik committed:
nix: update packaging to Qt 6

nix: disable warnings to errors due to bogus warning

Disable `-DWERROR` due to bogus warning

```
warning: More #if's than #endif's found
```

See: https://github.com/doxygen/doxygen/issues/11675

imincik commented:
`nix build .#qgis` currently fails on

```
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o -MF src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o -c /build/source/external/pdal_wrench/alg.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/alg.cpp:16:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [619/4081] Building CXX object src/providers/pdal/CMakeFiles/untwine.dir/__/__/__/external/untwine/epf/FileProcessor.cpp.o
qgis-unwrapped> [620/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o -MF src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o -c /build/source/external/pdal_wrench/main.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/main.cpp:25:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~


...



qgis-unwrapped> FAILED: python/gui/build/_gui/sip_guipart0.cpp python/gui/build/_gui/sip_guipart1.cpp python/gui/build/_gui/sip_guipart2.cpp python/gui/build/_gui/sip_guipart3.cpp python/gui/build/_gui/sip_guipart4.cpp python/gui/build/_gui/sip_guipart5.cpp python/gui/build/_gui/sip_guipart6.cpp python/gui/build/_gui/sip_guipart7.cpp python/gui/build/_gui/sip_guipart8.cpp python/gui/build/_gui/sip_guipart9.cpp python/gui/build/_gui/sip_guipart10.cpp python/gui/build/_gui/sip_guipart11.cpp python/gui/build/_gui/sip_guipart12.cpp python/gui/build/_gui/sip_guipart13.cpp python/gui/build/_gui/sip_guipart14.cpp python/gui/build/_gui/sip_guipart15.cpp python/gui/build/_gui/sip_guipart16.cpp python/gui/build/_gui/sip_guipart17.cpp python/gui/build/_gui/sip_guipart18.cpp python/gui/build/_gui/sip_guipart19.cpp python/gui/build/_gui/sip_guipart20.cpp python/gui/build/_gui/sip_guipart21.cpp python/gui/build/_gui/sip_guipart22.cpp python/gui/build/_gui/sip_guipart23.cpp python/gui/build/_gui/sip_guipart24.cpp /build/source/build/python/gui/build/_gui/sip_guipart0.cpp /build/source/build/python/gui/build/_gui/sip_guipart1.cpp /build/source/build/python/gui/build/_gui/sip_guipart2.cpp /build/source/build/python/gui/build/_gui/sip_guipart3.cpp /build/source/build/python/gui/build/_gui/sip_guipart4.cpp /build/source/build/python/gui/build/_gui/sip_guipart5.cpp /build/source/build/python/gui/build/_gui/sip_guipart6.cpp /build/source/build/python/gui/build/_gui/sip_guipart7.cpp /build/source/build/python/gui/build/_gui/sip_guipart8.cpp /build/source/build/python/gui/build/_gui/sip_guipart9.cpp /build/source/build/python/gui/build/_gui/sip_guipart10.cpp /build/source/build/python/gui/build/_gui/sip_guipart11.cpp /build/source/build/python/gui/build/_gui/sip_guipart12.cpp /build/source/build/python/gui/build/_gui/sip_guipart13.cpp /build/source/build/python/gui/build/_gui/sip_guipart14.cpp /build/source/build/python/gui/build/_gui/sip_guipart15.cpp /build/source/build/python/gui/build/_gui/sip_guipart16.cpp /build/source/build/python/gui/build/_gui/sip_guipart17.cpp /build/source/build/python/gui/build/_gui/sip_guipart18.cpp /build/source/build/python/gui/build/_gui/sip_guipart19.cpp /build/source/build/python/gui/build/_gui/sip_guipart20.cpp /build/source/build/python/gui/build/_gui/sip_guipart21.cpp /build/source/build/python/gui/build/_gui/sip_guipart22.cpp /build/source/build/python/gui/build/_gui/sip_guipart23.cpp /build/source/build/python/gui/build/_gui/sip_guipart24.cpp
qgis-unwrapped> cd /build/source/build/python/gui && /nix/store/dx4bdrs7mq3jfviqhszrc7l35ps9kg64-cmake-3.31.7/bin/cmake -E echo && /nix/store/zrakxg5nz3q6hm2f71x667xh6yi6hl0q-python3.13-sip-6.10.0/bin/sip-build --no-protected-is-public --pep484-pyi --no-make --concatenate=25 --qmake=/nix/store/l9kvcx3wna1bla7xpy1629hawjqmna4y-qtbase-6.9.0/bin/qmake --include-dir=/build/source/build/python --include-dir=/nix/store/jaivzh3rf3jkx10bjphk3zlgqilahpmv-python3.13-pyqt6-6.9.0/lib/python3.13/site-packages/PyQt6/bindings --api-dir /build/source/build/python --include-dir=/nix/store/snky0wkvwxgybl3bz7qzagbky1sa8rbm-python3.13-qscintilla-qt5-2.14.1/lib/python3.13/site-packages/PyQt6/bindings && /nix/store/dx4bdrs7mq3jfviqhszrc7l35ps9kg64-cmake-3.31.7/bin/cmake -E touch /build/source/build/python/gui/build/_gui/sip_guipart0.cpp /build/source/build/python/gui/build/_gui/sip_guipart1.cpp /build/source/build/python/gui/build/_gui/sip_guipart2.cpp /build/source/build/python/gui/build/_gui/sip_guipart3.cpp /build/source/build/python/gui/build/_gui/sip_guipart4.cpp /build/source/build/python/gui/build/_gui/sip_guipart5.cpp /build/source/build/python/gui/build/_gui/sip_guipart6.cpp /build/source/build/python/gui/build/_gui/sip_guipart7.cpp /build/source/build/python/gui/build/_gui/sip_guipart8.cpp /build/source/build/python/gui/build/_gui/sip_guipart9.cpp /build/source/build/python/gui/build/_gui/sip_guipart10.cpp /build/source/build/python/gui/build/_gui/sip_guipart11.cpp /build/source/build/python/gui/build/_gui/sip_guipart12.cpp /build/source/build/python/gui/build/_gui/sip_guipart13.cpp /build/source/build/python/gui/build/_gui/sip_guipart14.cpp /build/source/build/python/gui/build/_gui/sip_guipart15.cpp /build/source/build/python/gui/build/_gui/sip_guipart16.cpp /build/source/build/python/gui/build/_gui/sip_guipart17.cpp /build/source/build/python/gui/build/_gui/sip_guipart18.cpp /build/source/build/python/gui/build/_gui/sip_guipart19.cpp /build/source/build/python/gui/build/_gui/sip_guipart20.cpp /build/source/build/python/gui/build/_gui/sip_guipart21.cpp /build/source/build/python/gui/build/_gui/sip_guipart22.cpp /build/source/build/python/gui/build/_gui/sip_guipart23.cpp /build/source/build/python/gui/build/_gui/sip_guipart24.cpp
qgis-unwrapped> pyproject.toml: line 10: using '[tool.sip.metadata]' to specify the project metadata is deprecated and will be removed in SIP v7.0.0, use '[project]' instead
qgis-unwrapped> Querying qmake about your Qt installation...
qgis-unwrapped> These bindings will be built: gui.
qgis-unwrapped> Generating the gui bindings...
qgis-unwrapped> sip-build: /build/source/build/python/gui/gui.sip: line 17: column 13: 'Qsci/qscimod6.sip' could not be found
```

jef-n commented:
Wild guess: I suppose this is caused by `-DGDAL_DEBUG`

imincik commented:
> Wild guess: I suppose this is caused by `-DGDAL_DEBUG`

Great thanks, looks related to https://github.com/qgis/QGIS/pull/61316 indeed.

imincik commented:
Setting `"-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_CXX_FLAGS=-UGDAL_DEBUG"` helped to fix the issue with pdal. Now failing on `Qsci/qscimod6.sip` which is caused by missing QScintilla Qt6 package in Nixpkgs.

imincik commented:
Qt 6 build [really requires](https://github.com/qgis/QGIS/blob/798f63fc3c0d2616a5fbc8f47139fbeb5db7c052/python/CMakeLists.txt#L134) sip abi 13.9.

imincik commented:
https://github.com/NixOS/nixpkgs/pull/424421 is updating sip abi to the required version and now everything looks working well.

nyalldawson commented:
> Setting "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_CXX_FLAGS=-UGDAL_DEBUG" helped to fix the issue with pdal

Is the intention that these builds are used for testing only, or for actual user work? The debug releases are MUCH slower then release builds, so if these are end-user focused then it'd be nice to find a different solution.

imincik commented:
> > Setting "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_CXX_FLAGS=-UGDAL_DEBUG" helped to fix the issue with pdal
> 
> Is the intention that these builds are used for testing only, or for actual user work? The debug releases are MUCH slower then release builds, so if these are end-user focused then it'd be nice to find a different solution.

Definitely, this debug build is only workaround and needs to be removed.

autra commented:
> Definitely, this debug build is only workaround and needs to be removed.

Actually, one of the main use case for a flake in this repo is dogfooding. With that in mind, it'd be nice to keep a debug build available, maybe with its own entry, like `packages.qgis-debug` for instance?

imincik commented:
> > Definitely, this debug build is only workaround and needs to be removed.
> 
> Actually, one of the main use case for a flake in this repo is dogfooding. With that in mind, it'd be nice to keep a debug build available, maybe with its own entry, like `packages.qgis-debug` for instance?

Build is now working without debug workaround. But we can add debug parameter to Nix packaging with default value set to `False`.

imincik commented:
Debug build `-DCMAKE_BUILD_TYPE=Debug` fails without setting  `-DCMAKE_CXX_FLAGS=-UGDAL_DEBUG` on

```
qgis-unwrapped> [3895/4081] Building CXX object src/providers/pdal/CMakeFiles/untwine.dir/__/__/__/external/lazperf/detail/field_rgb14.cpp.o
qgis-unwrapped> [3896/4081] Building CXX object src/providers/pdal/CMakeFiles/untwine.dir/__/__/__/external/lazperf/detail/field_point14.cpp.o
qgis-unwrapped> [3897/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o -MF
 src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/main.cpp.o -c /build/source/external/pdal_wrench/main.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/main.cpp:25:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3898/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o -MF
src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/alg.cpp.o -c /build/source/external/pdal_wrench/alg.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/alg.cpp:16:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3899/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/boundary.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/boundary.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/boundary.cpp.o
 -MF src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/boundary.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/boundary.cpp.o -c /build/source/external/pdal_wrench/boundary.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/boundary.cpp:26:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3900/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/clip.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/clip.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/clip.cpp.o -MF
 src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/clip.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/clip.cpp.o -c /build/source/external/pdal_wrench/clip.cpp
qgis-unwrapped> In file included from /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Polygon.hpp:39,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/clip.cpp:21:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3901/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/density.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/density.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/density.cpp.o
-MF src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/density.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/density.cpp.o -c /build/source/external/pdal_wrench/density.cpp
qgis-unwrapped> In file included from /build/source/external/pdal_wrench/vpc.hpp:18,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/density.cpp:23:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3902/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/info.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/info.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/info.cpp.o -MF
 src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/info.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/info.cpp.o -c /build/source/external/pdal_wrench/info.cpp
qgis-unwrapped> In file included from /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Polygon.hpp:39,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/info.cpp:21:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3903/4081] Building CXX object src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/merge.cpp.o
qgis-unwrapped> FAILED: src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/merge.cpp.o
qgis-unwrapped> /nix/store/pbqah1qk4b5y14fqinr1h8zvhqy71v81-gcc-wrapper-14.3.0/bin/g++ -DGDAL_DEBUG -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_USE_QSTRINGBUILDER -DSIP_VERSION=0x060a00 -I/build/source/build/src/providers/pdal/pdal_wrench_autogen/include -I/build/source/build -I/build/source/external/untwine/api -I/build/source/external -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wno-deprecated-copy -g -std=gnu++17 -fvisibility=hidden -MD -MT src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/merge.cpp.o -M
F src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/merge.cpp.o.d -o src/providers/pdal/CMakeFiles/pdal_wrench.dir/__/__/__/external/pdal_wrench/merge.cpp.o -c /build/source/external/pdal_wrench/merge.cpp
qgis-unwrapped> In file included from /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Polygon.hpp:39,
qgis-unwrapped>                  from /build/source/external/pdal_wrench/merge.cpp:22:
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp: In member function 'OGRGeometryHS* pdal::Geometry::getOGRHandle()':
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:77:24: error: cannot convert 'std::unique_ptr<OGRGeometry>::pointer' {aka 'OGRGeometry*'} to 'OGRGeometryH' {aka 'OGRGeometryHS*'} in return
qgis-unwrapped>    77 |     { return m_geom.get(); }
qgis-unwrapped>       |              ~~~~~~~~~~^~
qgis-unwrapped>       |                        |
qgis-unwrapped>       |                        std::unique_ptr<OGRGeometry>::pointer {aka OGRGeometry*}
qgis-unwrapped> /nix/store/nxwgw5178dqvm04p9x6ifid6v2l347b8-pdal-2.8.4/include/pdal/Geometry.hpp:43:7: note: class type 'OGRGeometry' is incomplete
qgis-unwrapped>    43 | class OGRGeometry;
qgis-unwrapped>       |       ^~~~~~~~~~~
qgis-unwrapped> [3904/4081] Building CXX object src/providers/virtualraster/CMakeFiles/provider_virtualraster.dir/qgsvirtualrasterprovider.cpp.o
qgis-unwrapped> ninja: build stopped: subcommand failed.
```

Mrmaxmeier commented on nix/package.nix:
```diff
@@ -13,7 +13,7 @@
+  qgis-unwrapped = qt6Packages.callPackage ./unwrapped.nix {
     withGrass = withGrass;
     withServer = withServer;
     withWebKit = withWebKit;
```
Seems like this is a stale argument after fe354cc977d7c349a0bc17070f02246fe15b25d1

imincik replied:
Great thanks. You are right with this change.

m-kuhn commented on nix/set-pyqt6-package-dirs.patch:
```diff
@@ -0,0 +1,49 @@
+   endif()
+   if(QSCI_SIP_DIR)
+-    set(SIP_EXTRA_OPTIONS ${SIP_EXTRA_OPTIONS} -I ${QSCI_SIP_DIR})
++    set(SIP_BUILD_EXTRA_OPTIONS ${SIP_BUILD_EXTRA_OPTIONS} --include-dir=${QSCI_SIP_DIR})
```
Is this something to upstream?

imincik committed:
nix: build qgis package and docs in flake check

imincik commented:
Moving back to draft.

With current master version, documentation build has started to fail.

```
nix build .#docs -L

qgis-api-documentation-master> /build/source/src/core/qgis.cpp:746: error: More #if's than #endif's found (might be in an included file). (warning treated as error, aborting now)
qgis-api-documentation-master> make[3]: *** [doc/CMakeFiles/apidoc.dir/build.make:8264: doc/api/html/index.qhp] Error 1
qgis-api-documentation-master> make[3]: *** Deleting file 'doc/api/html/index.qhp'
qgis-api-documentation-master> make[2]: *** [CMakeFiles/Makefile2:272: doc/CMakeFiles/apidoc.dir/all] Error 2
qgis-api-documentation-master> make[1]: *** [CMakeFiles/Makefile2:279: doc/CMakeFiles/apidoc.dir/rule] Error 2
qgis-api-documentation-master> make: *** [Makefile:218: apidoc] Error 2
```

Any ideas ?

imincik commented:
Also, `qgis.qch` file is not created any more when `-DGENERATE_QHP=True` is set.

imincik committed:
nix: build with libspatialindex 2.0.0

Required by https://github.com/qgis/QGIS/pull/63196

imincik commented:
> Also, `qgis.qch` file is not created any more when `-DGENERATE_QHP=True` is set.

Fixed now.

imincik commented:
> When building without `-DWERROR=True` build succeeds with following warnings:
> 
> ```
> qgis-api-documentation-master> /build/source/src/core/qgis.cpp:746: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/core/geometry/qgsgeometry.h:3340: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/qgsgui.cpp:576: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/qgsstyleitemslistwidget.cpp:578: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/qgsstyleitemslistwidget.h:215: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/attributetable/qgsfeaturefilterwidget_p.h:134: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/callouts/qgscalloutwidget.h:254: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/sensor/qgssensorwidget.h:158: warning: More #if's than #endif's found (might be in an included file).
> qgis-api-documentation-master> /build/source/src/gui/symbology/qgsstylemanagerdialog.h:482: warning: More #if's than #endif's found (might be in an included file).
> ```
> 
> Looks like [doxygen/doxygen#11675](https://github.com/doxygen/doxygen/issues/11675) .

Fixed now.

imincik committed:
nix: exclude more qgis source code files

Exclude more of qgis source code files from Nix package to limit down
amount of rebuilds.

m-kuhn commented:
@timlinux thanks for testing.
I'm absolutely in favor of merging this, I am still waiting for a feedback of the discussion here https://github.com/qgis/QGIS/pull/62574#discussion_r2343953291, once that's sorted out we can very well merge.

imincik committed:
nix: update pyqt6 patch

Author: Augustin Trancart <augustin.trancart@oslandia.com>

imincik commented:
Problem fixed in d60ebe2587fb40921c50827cbbc34d84273d3955 . Thanks @autra !

imincik commented:
> No problem, I'm currently testing removing completely this patch, because I think it should work without.

@autra , is it working without patch ?

m-kuhn commented:
Let's merge and handle follow ups as they come
